azure/storageservice/client.go

```diff
@@ -0,0 +1,136 @@
+package storageservice
+
+import (
+	"encoding/base64"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/MSOpenTech/azure-sdk-for-go/azure"
+)
+
+const (
+	azureStorageServiceListURL = "services/storageservices"
+	azureStorageServiceURL     = "services/storageservices/%s"
+
+	azureXmlns = "http://schemas.microsoft.com/windowsazure"
+
+	errBlobEndpointNotFound = "Blob endpoint was not found in storage serice %s"
+	errParamNotSpecified    = "Parameter %s is not specified."
+)
+
+//NewClient is used to instantiate a new StorageServiceClient from an Azure client
+func NewClient(self *azure.Client) *StorageServiceClient {
+	return &StorageServiceClient{client: self}
+}
+
+func (self *StorageServiceClient) GetStorageServiceList() (*StorageServiceList, error) {
+	storageServiceList := new(StorageServiceList)
+
+	response, err := self.client.SendAzureGetRequest(azureStorageServiceListURL)
+	if err != nil {
+		return nil, err
+	}
+
+	err = xml.Unmarshal(response, storageServiceList)
+	if err != nil {
+		return storageServiceList, err
+	}
+
+	return storageServiceList, nil
+}
+
+func (self *StorageServiceClient) GetStorageServiceByName(serviceName string) (*StorageService, error) {
+	if len(serviceName) == 0 {
+		return nil, fmt.Errorf(errParamNotSpecified, "serviceName")
+	}
+
+	storageService := new(StorageService)
+	requestURL := fmt.Sprintf(azureStorageServiceURL, serviceName)
+	response, err := self.client.SendAzureGetRequest(requestURL)
+	if err != nil {
+		return nil, err
+	}
+
+	err = xml.Unmarshal(response, storageService)
+	if err != nil {
+		return nil, err
+	}
+
+	return storageService, nil
+}
+
+func (self *StorageServiceClient) GetStorageServiceByLocation(location string) (*StorageService, error) {
```

Comment:
@hopetobelievein yeah that's the use case logic, though. we can have some other pkg or repo on top of this (like `vmutil`) to make it easy to create VMs using the REST calls here. having such implicit logic may also harm user's storage accounts. for example, portal only places vhds on the storage account it has created. however this code can easily contaminate user's other storage accounts which has nothing to do with the specific VM deployment.